src/corehost/cli/hostpolicy.cpp

```diff
@@ -81,6 +81,13 @@ int run(const arguments_t& args)
     std::vector<char> tpa_paths_cstr, app_base_cstr, native_dirs_cstr, resources_dirs_cstr, fx_deps, deps, clrjit_path_cstr, probe_directories;
     pal::pal_clrstring(probe_paths.tpa, &tpa_paths_cstr);
     pal::pal_clrstring(args.app_dir, &app_base_cstr);
+
+    //removing trailing directory seperator for app compat reasons
+    while (app_base_cstr.size() > 2 && app_base_cstr.end()[-2] == DIR_SEPARATOR )
```

Comment:
you can't do a pop_back, as the last character in the vector is null,
we are just doing for appcompat